### PR TITLE
installer: skip Experimental Options page, if empty

### PR DESCRIPTION
**Pull request by dscho**
As of a75602aa0 (installer: hide the "experimental" FSMonitor setting by default, 2024-07-17), the FSMonitor option is hidden from the options page unless the user had enabled it previously.

However, this then left the page blank (or: all options hidden), not a good user experience. So let's skip it if it would appear blank.

This fixes https://github.com/git-for-windows/git/issues/5231.

### CODE REVIEW

File: installer/install.iss [hunk 1]
```diff
@@ -423,6 +423,7 @@ const
     GP_BuiltinAddI     = 4;
     GP_EnablePCon      = 5;
     GP_EnableFSMonitor = 6;
+    GP_Max             = 6;
```
Comment (rimrul) [inline installer/install.iss:426]: This is a great idea, but the name is slightly confusing since we use the same `GP_` prefix for other things.

**Review by rimrul — APPROVED** [review]
The code looks great. We should at some point rework the `GP_` prefix recycling though to avoid confusion.

File: installer/install.iss (hunk 1 above)
Comment (dscho) [reply installer/install.iss]: I revamped this rather completely. Could you have another look, @rimrul?

**Review by mjcheetham — APPROVED** [review]
Good fix and nice QOL improvements too :-)

**Comment by dscho** [comment]
/add relnote bug Due to a bug introduced in the v2.47 cycle, [the installer showed an empty "experimental options" page](https://github.com/git-for-windows/git/issues/5231), which was [fixed](https://github.com/git-for-windows/build-extra/pull/578).

**Comment by jerryfleurival** [comment]
@dscho 
> As of [a75602a](https://github.com/git-for-windows/build-extra/commit/a75602aa089e9a119764187484c8fcb775750231) (installer: hide the "experimental" FSMonitor setting by default, 2024-07-17), the FSMonitor option is hidden from the options page unless the user had enabled it previously.
> 
> However, this then left the page blank (or: all options hidden), not a good user experience. So let's skip it if it would appear blank.
> 
> This fixes [git-for-windows/git#5231](https://github.com/git-for-windows/git/issues/5231).

Yes, this indeed need to improve user experience of the last page. As for all, This does not effect the installation process or nothing wrong with the installation as you can go ahead and just hit the "Install" button and installation will begin. Though, at some point in the future, this (user experience) needs to be address. 

As for enabling experimental options, I would assume the user set `core.fsmonitor` to `true`  either  by system, global or repository(local) levels  in the command line themselves. Experimental option such as `fsmonitor.daemon`.

**Comment by dscho** [comment]
> > This fixes [git-for-windows/git#5231](https://github.com/git-for-windows/git/issues/5231).
> 
> Yes, this indeed need to improve user experience of the last page.

@jerryfleurival I don't understand... Do you mean to say that this is _not_ fixed for you? For me, the last page is:

image

After that, I only get the following page if I have a Git Bash open, otherwise it immediately starts to install Git for Windows:

image

At no point do I see an empty "Experimental Options" page. Is this different for you?

**Comment by jerryfleurival** [comment]
@dscho I'm on version 2.47.0.2. I left those two unchecked, and then came a blank Experimental Options page. 
Ref: https://github.com/git-for-windows/git/issues/5273#issue-2687957789

**Comment by dscho** [comment]
> I'm on version 2.47.0.2.

Well, the [fix](https://github.com/git-for-windows/git/issues/5231) is [in v2.47.1](https://github.com/git-for-windows/git/milestone/95?closed=1), that's why you don't have it. You need to upgrade.

**Pull request merged**